### PR TITLE
feat: Making contact generation optional based on shape types

### PR DESCRIPTION
**Pull request by Rhuagh**
This change is [Reviewable](https://reviewable.io/reviews/rhuagh/rhusics/38)

### CODE REVIEW

File: src/collide/narrow.rs [hunk 1]
```diff
@@ -11,15 +11,15 @@ use collision::algorithm::minkowski::{SimplexProcessor, EPA, GJK};
...
+use collide::{Collider, CollisionMode, CollisionShape};
 
 /// Base trait implemented by all narrow phase algorithms.
 ///
 /// # Type parameters:
 ///
 /// - `P`: collision primitive type
 /// - `T`: model-to-world transform type
-pub trait NarrowPhase<P, T>: Send
+pub trait NarrowPhase<P, T, Y>: Send
```
Comment (torkleyy) [inline src/collide/narrow.rs]: Maybe let Y default to () so you don't have to specify it.

**Comment by Rhuagh** [comment]
Review status: 0 of 15 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/collide/narrow.rs, line 22 at r1](https://reviewable.io:443/reviews/rhuagh/rhusics/38#-L01OgcBfCj2mYUpajRQ-r1-22:-L03v96M5EeChOY3f8_D:b-896fix) ([raw file](https://github.com/rhuagh/rhusics/blob/044b7a672a8e94875fed53f302cd5404f6bd1009/src/collide/narrow.rs#L22)):*
<details><summary><i>Previously, torkleyy (Thomas Schaller) wrote…</i></summary><blockquote>

Maybe let Y default to () so you don't have to specify it.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/rhuagh/rhusics/38)*

File: src/collide/narrow.rs [hunk 2]
```diff
@@ -19,7 +19,7 @@ use collide::{Collider, CollisionMode, CollisionShape};
 ///
 /// - `P`: collision primitive type
 /// - `T`: model-to-world transform type
-pub trait NarrowPhase<P, T, Y>: Send
+pub trait NarrowPhase<P, T, Y = ()>: Send
```
Comment (Xaeroxe) [inline src/collide/narrow.rs]: Add Y to documentation please

**Comment by Rhuagh** [comment]
Review status: 0 of 15 files reviewed at latest revision, 2 unresolved discussions.

---

*[src/collide/narrow.rs, line 22 at r2](https://reviewable.io:443/reviews/rhuagh/rhusics/38#-L01OgcBfCj2mYUpajRQ-r2-22:-L06SDfUHQmx8X3zN34r:b-896fix) ([raw file](https://github.com/rhuagh/rhusics/blob/f938f148e983f8a15f3952159c31d4d7c1050c19/src/collide/narrow.rs#L22)):*
<details><summary><i>Previously, Xaeroxe (Jacob Kiesel) wrote…</i></summary><blockquote>

Add Y to documentation please
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/rhuagh/rhusics/38)*

**Pull request merged**